cpp/src/arrow/array/array_dict_test.cc

```diff
@@ -22,6 +22,8 @@
 #include <string>
 #include <vector>
 
+#include <iostream>
```

Comment:
Is this necessary? If so, please put this include with the standard library includes above.

Reply:
This should be removed before committing but I forgot.
Thanks for catching this!